src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard-v3/dashboard-area-chart/dashboard-area-chart.component.ts

```diff
@@ -220,45 +250,47 @@ export class DashboardAreaChartComponent implements OnInit, OnChanges, AfterView
   private setChartTicks() {
-    if (this.chart && this.maxValue) {
-      let [maxValue, maxValueDataUnits] = this.convertUnits(this.maxValue).split(' ');
-      this.chart.chart.options.scales.yAxes[0].ticks.suggestedMax = maxValue;
-      this.chart.chart.options.scales.yAxes[0].ticks.suggestedMin = 0;
-      this.chart.chart.options.scales.yAxes[0].ticks.stepSize = Number((maxValue / 2).toFixed(0));
-      this.chart.chart.options.scales.yAxes[0].ticks.callback = (value: any) => {
-        if (value === 0) {
-          return null;
-        }
-        return this.fillString(`${value} ${maxValueDataUnits}`);
-      };
-      this.chart.chart.update();
-    } else if (this.chart && this.data) {
+    if (this.chart) {
       let maxValue = 0,
         maxValueDataUnits = '';
-      let maxValueData = Math.max(...this.data.map((values: any) => values[1]));
-      if (this.data2) {
-        var maxValueData2 = Math.max(...this.data2.map((values: any) => values[1]));
-        [maxValue, maxValueDataUnits] = this.convertUnits(
-          Math.max(maxValueData, maxValueData2)
-        ).split(' ');
-      } else {
-        [maxValue, maxValueDataUnits] = this.convertUnits(Math.max(maxValueData)).split(' ');
-      }
-
-      this.chart.chart.options.scales.yAxes[0].ticks.suggestedMax = maxValue * 1.2;
-      this.chart.chart.options.scales.yAxes[0].ticks.suggestedMin = 0;
-      this.chart.chart.options.scales.yAxes[0].ticks.stepSize = Number(
-        ((maxValue * 1.2) / 2).toFixed(0)
-      );
-      this.chart.chart.options.scales.yAxes[0].ticks.callback = (value: any) => {
-        if (value === 0) {
-          return null;
-        }
-        if (!maxValueDataUnits) {
-          return this.fillString(`${value}`);
+      if (this.maxValue) {
+        [maxValue, maxValueDataUnits] = this.convertUnits(this.maxValue).split(' ');
+        this.chart.chart.options.scales.yAxes[0].ticks.suggestedMax = maxValue;
+        this.chart.chart.options.scales.yAxes[0].ticks.suggestedMin = 0;
+        this.chart.chart.options.scales.yAxes[0].ticks.stepSize = Number((maxValue / 2).toFixed(0));
+        this.chart.chart.options.scales.yAxes[0].ticks.callback = (value: any) => {
+          if (value === 0) {
+            return null;
+          }
+          return this.fillString(`${value} ${maxValueDataUnits}`);
+        };
+      } else if (this.data) {
+        let maxValueData = Math.max(...this.data.map((values: any) => values[1]));
+        if (this.data2) {
+          var maxValueData2 = Math.max(...this.data2.map((values: any) => values[1]));
+          [maxValue, maxValueDataUnits] = this.convertUnits(
+            Math.max(maxValueData, maxValueData2)
+          ).split(' ');
+        } else {
+          [maxValue, maxValueDataUnits] = this.convertUnits(maxValueData).split(' ');
         }
-        return this.fillString(`${value} ${maxValueDataUnits}`);
-      };
+
+        this.chart.chart.options.scales.yAxes[0].ticks.suggestedMax = maxValue * 1.2;
```

Comment:
Let's not hard-code literals... What if you find that 1.2 is too much or not enough? You have to replace every occurrence... And it's good to read an explanation of why you decided to go with a 20% head room (in case someone else needs to fix this in the future). Let's be kind to our future reviewers (it might be yourself in 6 months):
```suggestion
        this.chart.chart.options.scales.yAxes[0].ticks.suggestedMax = maxValue * EXTRA_ROOM;
```

Reply:
Right. Fixed it.